[CUMULUS-1544] Add documentation for bulk granule endpoint (#275)
nasa/cumulus-api: content/granules.md

  "detail": "Record deleted"
}
```

## Bulk Operations

> Apply a workflow to the Granules provided. Granule can be sent as a list of IDs or as an Elasticsearch query to the Metrics' Elasticsearch.

Reviewer: ```suggestion
Apply a workflow to the granules provided. Granule can be sent as a list of IDs or as an Elasticsearch query to the configured metrics Elasticsearch. To use Elasticsearch, the following variables must be configured in your deployment: `metrics_es_host`, `metrics_es_username`, and `metrics_es_password`.
```

Overview of the request fields:

> | Field | Required | Value | Description |

Reviewer: This looks really good.

| --- | --- | --- | --- |
| `workflow` | `Y` | `string` | Worfklow to be applied to all granules |
| `queueName` | `N` | `string` | Queue to process Granules in |

> | `ids` | `Y` - if no query | `Array` | List of IDs to process. Required if there is no Elasticsearch query provided |

Reviewer: ```suggestion
| `ids` | `Y` - if no query | `Array<string>` | List of IDs to process. Required if there is no Elasticsearch query provided |
```

| `query` | `Y` - if no ids | `Object` | Query to Elasticsearch to determine which Granules to go through given workflow. Required if no IDs are given. |
| `index` | `Y` - if query is present | `string` | Elasticsearch index to search with the given query |

```endpoint
POST /granules/bulk
```

#### Example requests

> Request with Elasticsearch query generated by Kibana:

Reviewer: This doesn't get formatted right so instead of putting it all under `Example Requests` I'd do:

```
#### Example request with Elasticsearch query generated by Kibana

<request>

#### Example request with given Granule IDs:

<request>

#### Example response

<response>
```

```curl

> $ curl --request POST https://example.com/dev/granules/bulk --header 'Authorization: Bearer ReplaceWithTheToken' 

Reviewer: ```suggestion
$ curl --request POST https://example.com/granules/bulk --header 'Authorization: Bearer ReplaceWithTheToken' 
```

>   --data '{"workflowName": "HelloWorldWorkflow", index: "index-in-es", query": {"size": 500, "query": {"filter": [{"bool": { "filter": [{"bool": {"should": [{"match": {"granuleId": "MOD09GQ.A2016358.h13v04.006.2016360104606"}}], "minimum_should_match": 1}}, {"bool": {"should": [{"match": {"status": "FAILED"}}],"minimum_should_match": 1}}]}}],"should": [],"must_not": []}}}'

Author: I cut out some of the Kibana query to help make this more readable but can add it back if we want more accuracy

Reviewer: I think it's good, I just think format the JSON though and you're good to go

```

Request with given Granule IDs:

```curl
curl -X POST

>   https://example.com:8000/dev/granules/bulk --header 'Authorization: Bearer ReplaceWithTheToken' --data '{"ids": ["MOD09GQ.A2016358.h13v04.006.2016360104606"], "workflowName": "HelloWorldWorkflow"}'

Reviewer: ```suggestion
  https://example.com/granules/bulk --header 'Authorization: Bearer ReplaceWithTheToken' --data '{"ids": ["MOD09GQ.A2016358.h13v04.006.2016360104606"], "workflowName": "HelloWorldWorkflow"}'
```
